llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -24414,6 +24414,10 @@ bool RISCVTargetLowering::isLegalSpeculativeLoad(EVT DataType,
   if (!isLegalElementTypeForRVV(ScalarType))
     return false;
 
+  if (!Subtarget.enableUnalignedVectorMem() &&
+      Alignment < ScalarType.getStoreSize())
```

Comment:
Do we have a test for an alignment that's too small?

Reply:
Added. Thanks!